drivers/timer/cortex_m_systick.c

```diff
@@ -749,14 +749,15 @@ u32_t _timer_cycle_get_32(void)
 #ifdef CONFIG_TICKLESS_KERNEL
 return (u32_t) get_elapsed_count();
 #else
-	u32_t cac, count;
+	u64_t cac;
+	u32_t count;
 
 	do {
 		cac = clock_accumulated_count;
 		count = SysTick->LOAD - SysTick->VAL;
 	} while (cac != clock_accumulated_count);
 
-	return cac + count;
+	return (u32_t)(cac + count);
```

Comment:
Since we're going to perform 64-bit arithmetic anyway shouldn't we implement this function as
```
return (u32_t)_timer_cycle_get();
```